pkg/common/log/logger.go

```diff
@@ -4,58 +4,75 @@ import (
 	"fmt"
 	"io/ioutil"
 	"os"
+	"strings"
+	"sync"
 	"time"
 
-	"github.com/ServiceComb/paas-lager"
+	"github.com/go-mesh/openlogging"
+	"github.com/kubeedge/beehive/pkg/common/util"
+
+	log "github.com/ServiceComb/paas-lager"
 	"github.com/ServiceComb/paas-lager/rotate"
 	"github.com/go-chassis/paas-lager/third_party/forked/cloudfoundry/lager"
-	"gopkg.in/yaml.v2"
+	yaml "gopkg.in/yaml.v2"
+)
 
-	"github.com/kubeedge/beehive/pkg/common/config"
+const (
+	LOGCONFIGPATH = "/conf/logging.yaml"
 )
 
 // LOGGER var
 var LOGGER lager.Logger
-
-func init() {
-
-	logConfig := &log.Config{
-		LoggerLevel:   "DEBUG",
-		EnableRsyslog: false,
-		LogFormatText: true,
-		Writers:       []string{"stdout"},
-	}
-
-	filename := config.GetConfigDirectory() + "/conf/logging.yaml"
-
-	fmt.Printf(filename)
-	_, err := os.Stat(filename)
-	if err == nil {
-		content, err := ioutil.ReadFile(filename)
-		if err != nil {
-			fmt.Printf("Got error when reading yaml config file:%v\n", err)
-		}
-		err = yaml.Unmarshal(content, logConfig)
-		if err != nil {
-			fmt.Printf("Got error when reading yaml config file:%v\n", err)
+var once = sync.Once{}
+
+func InitializeLogger() {
+	once.Do(func() {
+		logConfig := &log.Config{
+			LoggerLevel:   "DEBUG",
+			EnableRsyslog: false,
+			LogFormatText: true,
+			Writers:       []string{"stdout"},
 		}
-		fmt.Printf("logConfig:%v\n", logConfig)
-	} else {
-		fmt.Printf("Got error when reading yaml config file:%v\n", err)
-	}
 
-	log.Init(*logConfig)
+		filename := strings.TrimSuffix(GetConfigDir(), "/") + LOGCONFIGPATH
+
+		//fmt.Println("logging.yaml's path is " + filename)
+		_, err := os.Stat(filename)
+		if err == nil {
+			content, err := ioutil.ReadFile(filename)
+			if err != nil {
+				fmt.Printf("Got error when reading yaml config file:%v\n", err)
```

Comment:
add a return here ?
Applicable to all similar err cases

Reply:
this method is calling in the init method which can not return any errors anymore. I think log the error is enough since the following logic may fails if we got error here